subsys/mgmt/mcumgr/grp/enum_mgmt/src/enum_mgmt.c

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) - 2024 Extreme Engineering Solutions
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/debug/object_tracing.h>
+#include <zephyr/kernel_structs.h>
+#include <zephyr/mgmt/mcumgr/mgmt/mgmt.h>
+#include <zephyr/mgmt/mcumgr/smp/smp.h>
+#include <zephyr/mgmt/mcumgr/mgmt/handlers.h>
+#include <zephyr/mgmt/mcumgr/grp/enum_mgmt/enum_mgmt.h>
+#include <zephyr/logging/log.h>
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <zcbor_common.h>
+#include <zcbor_encode.h>
+#include <zcbor_decode.h>
+
+#include <mgmt/mcumgr/util/zcbor_bulk.h>
+
+static int enumeration_mgmt_list(struct smp_streamer *ctxt)
+{
+	int count;
+	zcbor_state_t *zse = ctxt->writer->zs;
+	bool ok;
+
+	count = mgmt_get_num_groups();
+	int ids[count];
+
+	count = mgmt_get_group_ids(ids);
+
+	if (count == 0) {
+		return MGMT_ERR_ENOTSUP;
+	}
+
+	ok = zcbor_tstr_put_lit(zse, "groups") &&
+		zcbor_list_start_encode(zse, count);
+
+	for (int i = 0; i < count; ++i) {
+		ok = zcbor_uint32_put(zse, ids[i]);
+		if (!ok) {
+			return MGMT_ERR_EBADSTATE;
+		}
+	}
+
+	ok = zcbor_list_end_encode(zse, count);
+
+	return ok ? MGMT_ERR_EOK : MGMT_ERR_ECORRUPT;
+}
+
+static int enumeration_mgmt_info(struct smp_streamer *ctxt)
+{
+	zcbor_state_t *zse = ctxt->writer->zs
+	bool ok;
+
+	ok = zcbor_tstr_put_lit(zse, "name") &&
+		zcbor_tstr_put_lit(zse, ENUMERATION_MGMT_NAME) &&
+		zcbor_tstr_put_lit(zse, "rev") &&
+		zcbor_uint32_put(zse, ENUMERATION_MGMT_VER);
```

Comment:
no, the implementation of a function needs to be compatible, it does not return a name or version

Reply:
To what does this need to be compatible? I'm defining a new command that explicitly is for the purpose of returning the name and version of this group.

Reply:
I'm not sure what you mean by "random index". My rationale for including this command (and this PR in general) is to enable a more dynamic implementation of MCUmgr.  A remote system communicating with the SoC could discover the registered groups, along with their purposes, without the need for static knowledge of the SoC's implementation. This is less useful for the lower commands, I understand, but I add it with an eye toward user-defined groups.

Reply:
Why is this command needed? This group would occupy group 10, so group 10 command 0 would be for getting a list of support groups - that's all fine, why do you then need another command to return "enumeration" and "1"?

Reply:
I can eliminate it for this group. For background, the need for this change was discovered in the course of implementing user-specific groups. It was useful for each of those groups to have a name and revision, since it wasn't a guarantee that each system would use the same IDs for the same groups, or that the implementation for a group might not change across revisions. I agree that it's not important for a group that's defined by the standard (as this would be), it was just included as an artifact of the development process.